pkg/sql/alter_table.go

```diff
@@ -306,6 +307,66 @@ func (n *alterTableNode) startExec(params runParams) error {
 					}
 				}
 			}
+
+			// Drop check constraints which reference column
+			// For every check on the table, walk the check expr
+			// and ensure any check which references the column
+			// is removed from the table descriptor.
+			validChecks := n.tableDesc.Checks[:0]
+			shouldDrop := map[int]bool{}
+
+			for i, check := range n.tableDesc.Checks {
+				expr, err := parser.ParseExpr(check.Expr)
+				if err != nil {
+					return err
+				}
+
+				preFn := func(expr tree.Expr) (err error, recurse bool, newExpr tree.Expr) {
+					vBase, ok := expr.(tree.VarName)
+					if !ok {
+						// Not a VarName, don't do anything to this node.
+						return nil, true, expr
+					}
+
+					v, err := vBase.NormalizeVarName()
+					if err != nil {
+						return err, false, nil
+					}
+
+					c, ok := v.(*tree.ColumnItem)
+					if !ok {
+						return nil, true, expr
+					}
+
+					foundCol, err := n.tableDesc.FindActiveColumnByName(string(c.ColumnName))
```

Comment:
You should be able to skip this and compare `c.ColumnName` against `col.Name` directly.